tools/testsys/src/run.rs

```diff
@@ -0,0 +1,230 @@
+            (Some(metadata_url), Some(targets_url), _) => Some(TufRepoConfig {
+                metadata_url,
+                targets_url,
+            }),
+            (None, None, Some(repo)) => {
+                if let (Some(metadata_base_url), Some(targets_url)) =
+                    (&repo.metadata_base_url, &repo.targets_url)
+                {
+                    Some(TufRepoConfig {
+                        metadata_url: format!(
+                            "{}{}/{}",
+                            metadata_base_url, &self.variant, &self.arch
+                        ),
+                        targets_url: targets_url.to_string(),
+                    })
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        };
+
+        match variant.family() {
+            "aws-k8s" => {
+                let file = File::open(&self.ami_input).context("Unable to open amis.json")?;
+                let ami_input: HashMap<String, Image> = serde_json::from_reader(file)
+                    .context(format!("Unable to deserialize '{}'", self.ami_input))?;
+                ensure!(!ami_input.is_empty(), "amis.json is empty");
+                let bottlerocket_ami = &ami_input
+                    .get(&region)
+                    .context(format!("ami not found for region '{}'", region))?
+                    .id;
+                let aws_k8s = AwsK8s {
+                    arch: self.arch,
+                    variant: self.variant,
+                    region,
+                    assume_role: self.assume_role,
+                    instance_type: self.instance_type,
+                    ami: bottlerocket_ami.to_string(),
+                    secrets,
+                    kube_conformance_image: None,
+                    _repo_config: repo_config,
+                };
+                let crds = aws_k8s.create_crds(self.test_flavor, &TestsysImages::new_from_env())?;
+                for crd in crds {
+                    println!("CRD:\n{:?}", crd);
+                    let crd = client
+                        .create_object(crd)
+                        .await
+                        .context("Unable to create object")
+                        .unwrap();
+                    println!("Done");
+                    println!("Successfully added '{:#?}'", crd.name());
+                }
+            }
+            other => return Err(anyhow!("Unrecognized variant '{}'", other)),
+        };
+
+        Ok(())
+    }
+}
+
+fn parse_key_val(s: &str) -> Result<(String, SecretName)> {
+    let mut iter = s.splitn(2, '=');
+    let key = iter.next().context("Key is missing")?;
+    let value = iter.next().context("Value is missing")?;
+    Ok((key.to_string(), SecretName::new(value)?))
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum TestType {
+    /// Run all available tests on a given arch and variant (certified conformance, migrations, ipv6)
```

Comment:
I don't think we have figured out how to support IPV6 just yet.

```suggestion
    /// Run all available tests on a given arch and variant (certified conformance, migrations)
```